### ISSUE TITLE
Can not register handler edomh:// when data path contain national letters. 

### ISSUE THREAD

**Issue by SirAdams** [jixxed/ed-odyssey-materials-helper#312]
When i have on my disk national letters, in utf-8, i can not register handler. 
If i move folder to place without national letter then registration work. 
In windows registry national letter is change to two different characters.

**Comment by jixxed** [comment]
Looks like the app needs to use UTF-16LE for the registry. Can add a fix for the next release, do you have an example I can use to reproduce?

**Comment by SirAdams** [comment]
Yep. "Gry-Narzędzia"

**Comment by jixxed** [comment]
I did not have any issue with it on UTF-8 mode. added the fix nonetheless in 2.66, so let me know if it fixed your issue

**Comment by SirAdams** [comment]
I tested it with version 2.69 and the problem does not occur. Thank you.

**Comment by SirAdams** [comment]
I apologize for not writing back sooner. 

Unfortunately the problem has not been solved and I do not know why. 
This is my full directory path:
"D:\Gry-Narzędzia\Elite.Dangerous.Odyssey.Materials.Helper.portable"

I misspelled because the Windows code page is Windows-1250.

Unfortunately, I do not know why, but if the program is in this location then registering the link is not possible. 
I have checked the permissions of the directory, including giving a record for all, but this did not help. 
At least the program says it is not registered and without that I can't link it to my account. 

Maybe the procedure for checking if the link is registered is working incorrectly?

**Comment by SirAdams** [comment]
Today I decided to do another test and changed the "ę" to "e" and found that I could register and unregister the program without any problems. 
I also did a test with another national character and it doesn't work either. So registering and unregistering works if there are no national characters in the path.

**Comment by SirAdams** [comment]
I include log from ProcessMonitor
[Logfile.zip](https://github.com/user-attachments/files/15524677/Logfile.zip)

**Comment by jixxed** [comment]
As the registry is updated correctly, there won't be any issue there anymore. I will add some logging to the next version that will write out the registry value as the application reads it and what it expects. there may be an issue with the readout of the registry to see whether the application is registered.
I am still unable to reproduce it locally. the registry is utf-16le and code is utf-8, but for me that doesn't seem to matter, so it might be something in windows itself.

**Comment by jixxed** [comment]
I do have this enabled, which might be the reason why it works for me
https://exploratory.io/note/exploratory/Enabling-UTF-8-on-Windows-hYc3yWL0

**Comment by SirAdams** [comment]
Yes, after enabling UTF-8 in Windows, the program registers correctly.

**Comment by PancakeNM** [comment]
Can't register too... Tried already everything written here but still nothing helps.
Here's support package if this helps
[support.2024-08-24_13.19.25.961.zip](https://github.com/user-attachments/files/16736219/support.2024-08-24_13.19.25.961.zip)

**Comment by jixxed** [comment]
I see the following error: java.io.IOException: Cannot run program "powershell": CreateProcess error=2, Не удается найти указанный файл

Do you have powershell installed? Should come standard with W10/W11

**Comment by SirAdams** [comment]
I think that enabling UTF-8 is not the solution to the problem because I lose national letters in other programs. Perhaps if I would have done it right after installing windows 10....
I think there is a better solution for this.

**Comment by PancakeNM** [comment]
> I see the following error: java.io.IOException: Cannot run program "powershell": CreateProcess error=2, Не удается найти указанный файл
> 
> Do you have powershell installed? Should come standard with W10/W11

Yes i have. I even tried to reinstall it just now but that didn't helped

**Comment by jixxed** [comment]
If registering the app to handle edomh urls fails, you can also try this registry script as long as the app is installed in the default location:

Save the following contents to a file named `edomh.reg` and then doubleclick it to add the contents to the registry folder.
```
Windows Registry Editor Version 5.00

[HKEY_CLASSES_ROOT\edomh]
"URL Protocol"=""

[HKEY_CLASSES_ROOT\edomh\shell]

[HKEY_CLASSES_ROOT\edomh\shell\open]

[HKEY_CLASSES_ROOT\edomh\shell\open\command]
@="\"%USERPROFILE%\\AppData\\Local\\Elite Dangerous Odyssey Materials Helper Launcher\\program\\Elite Dangerous Odyssey Materials Helper.exe\" \"%1\""
""=hex(2):22,00,25,00,55,00,53,00,45,00,52,00,50,00,52,00,4f,00,46,00,49,00,\
  4c,00,45,00,25,00,5c,00,41,00,70,00,70,00,44,00,61,00,74,00,61,00,5c,00,4c,\
  00,6f,00,63,00,61,00,6c,00,5c,00,45,00,6c,00,69,00,74,00,65,00,20,00,44,00,\
  61,00,6e,00,67,00,65,00,72,00,6f,00,75,00,73,00,20,00,4f,00,64,00,79,00,73,\
  00,73,00,65,00,79,00,20,00,4d,00,61,00,74,00,65,00,72,00,69,00,61,00,6c,00,\
  73,00,20,00,48,00,65,00,6c,00,70,00,65,00,72,00,20,00,4c,00,61,00,75,00,6e,\
  00,63,00,68,00,65,00,72,00,5c,00,70,00,72,00,6f,00,67,00,72,00,61,00,6d,00,\
  5c,00,45,00,6c,00,69,00,74,00,65,00,20,00,44,00,61,00,6e,00,67,00,65,00,72,\
  00,6f,00,75,00,73,00,20,00,4f,00,64,00,79,00,73,00,73,00,65,00,79,00,20,00,\
  4d,00,61,00,74,00,65,00,72,00,69,00,61,00,6c,00,73,00,20,00,48,00,65,00,6c,\
  00,70,00,65,00,72,00,2e,00,65,00,78,00,65,00,22,00,20,00,22,00,25,00,31,00,\
  22,00,00,00
```

**Comment by MegaShinySnivy** [comment]
Has there been any progress on this? I'm in a similar spot and none of the fixes have helped with getting EDOMH to see these links, and Coriolis doesn't allow you to copy and paste the link in question

**Comment by jixxed** [comment]
There is no progress to be made on this really. It is unclear why some systems have issues. It could be some security software blocking changes to the registry or some kind of regional setting. Might not even have the same cause for everyone.

So far the registry script worked for everyone as a workaround. 
Some browsers can still have issues (firefox mostly)

here is a link for a blank wishlist if you need to try one:
edomh://horizonswishlist/?eNpFzDEKhTAQRdG9TG1A_RpJtiK_GM0LDmgEE7UQ926wsb0c7kU4EBJZOiVOs8REBR3YoqyBbFV82V607-KyHFqj3eCgSu9_quGuVsw1lPfOlEZDj5rzJfCCrBPepyQskWz_v-8HqHslBg==

You can stick it in the Run Command window, the browser address bar and even the Windows Explorer address bar
